src/submission/submission-to-clinical/submission-to-clinical.ts

```diff
@@ -416,7 +422,16 @@ const sendMessageOnUpdatesFromClinicalSubmission = async (
     );
 
   if (submissionHasProgramUpdates) {
-    await messenger.getInstance().sendProgramUpdatedMessage(submission.programId);
+    // use model to fetch donorIds for updated donors from their donorSubmitterIds
+    const programId = submission.programId;
+    const donorSubmitterIds = Object.entries(submission.clinicalEntities)
+      .filter(([_, entity]) => entity.stats.new.length > 0 || entity.stats.updated.length > 0)
+      .map(([_, entity]) => entity.dataUpdates.map(update => update.info.donorSubmitterId))
+      .reduce((acc, curr) => acc.concat(curr), []);
+    const donors = await donorDao.findByProgramAndSubmitterIds(programId, donorSubmitterIds);
+    const donorIds = donors ? donors.filter(d => d.donorId).map(d => String(d.donorId)) : [];
```

Comment:
Is there a reason we need to filter donors here? We shouldn't ever have a donor without a donorId...